include/modem/modem_jwt.h

```diff
@@ -76,6 +76,20 @@ struct jwt_data {
  */
 int modem_jwt_generate(struct jwt_data *const jwt);
 
+/**
+ * @brief Fetch modem UUID.
+ *
+ * This helper function utilizes JWT library to decode UUID from
+ * modem generated tokens. If succesfull a string is returned that
+ * contains UUID. Any error leads to this function returning NULL.
+ *
+ * Returned string consumes memory, which must later on be freed with
+ * @ref modem_jwt_free.
+ *
+ * @return string containing UUID or NULL on failure.
+ */
+const char *modem_jwt_get_uuid();
```

Comment:
could you make the function signature like this one: https://github.com/nrfconnect/sdk-nrf/blob/8f5922cb5cd660612975d8226e5b52d3d318cce7/include/modem/modem_attest_token.h#L128

that way the user can get the device uuid and/or the fw uuid.  also the user won't need to free memory.
you can re-use the UUID structs/defines.